### DISCUSSION TITLE
How the InputPreset in oboe mapping to MediaRecorder.AudioSource in the Android Java API

### DISCUSSION THREAD

**Discussion by xzhan96** [google/oboe#1775] [General]
I See the enum define of  InputPreset in Definitions.h: "Note that these match the equivalent values in MediaRecorder.AudioSource in the Android Java API."

However, the function OpenSLES_convertInputPreset in AudioInputStreamOpenSLES.cpp mapping both InputPreset::VoiceRecognition and  InputPreset::VoicePerformance into SL_ANDROID_RECORDING_PRESET_VOICE_RECOGNITION
https://cs.android.com/android/platform/superproject/+/master:external/oboe/src/opensles/AudioInputStreamOpenSLES.cpp;l=40?q=OpenSLES_convertInputPreset

I wondering why? if the real implementation of them in OpenSL are same?

**Comment by philburk** [comment]
OpenSL is deprecated.
There is no equivalent to VoicePerformance in OpenSL ES and we do not plan to add it.
So we map VoicePerformance to the nearest InputPreset that OpenSL ES supports.
